main.c

```diff
@@ -655,6 +656,9 @@ static int dev_added(struct tcmu_device *dev)
 	tcmu_dev_dbg(dev, "Got block_size %ld, size in bytes %lld\n",
 		     block_size, dev_size);
 
+	write_cache_enabled = tcmu_get_attribute(dev, "emulate_write_cache");
```

Comment:
Let's actually drop the write cache handling from this patchset or break it out into separate patches at least.

The problem is if the user requested write cache disabled, for most likely safety reasons, there is only qcow able to handle it in these patches. I can convert glfs and rbd for us (they always want write cache disabled so we can just print an error if someone tries to enable it). I assume you can take file_optical. Either one of us can take file_example.